docs/usage/jssprovider.md

```diff
@@ -0,0 +1,205 @@
+ - `noModDB`: Don't open the security module DB, just initialize the PKCS #11
+   module. `boolean`, defaults to `false`.
+ - `noPK11Finalize`: never call `C_Finalize` on any PKCS#11 module. `boolean`,
+   defaults to `false`.
+ - `noRootInit`: Don't try to look for the root certs module automatically.
+   `boolean`, defaults to `false`.
+ - `ocspCheckingEnabled`: To have NSS check the OCSP responder for when
+   verifying certificates, set this flags to `true`. `boolean`, defaults to
+   `false`.
+ - `ocspResponderCertNickname`: The nickname of the cert to trust (expected)
+   to sign the OCSP responses. `String`, defaults to `null`.
+ - `ocspResponderURL`: Specify the location and cert of the responder.
+   `String`, defaults to `null`.
+ - `optimizeSpace`: Use smaller tables and caches. `boolean`, defaults to
+   `false`.
+ - `PK11Reload`: Init `PK11Reload` to ignore the
+   `CKR_CRYPTOKI_ALREADY_INITIALIZED` error when loading PKCS#11 modules.
+   `boolean`, defaults to `false`.
+ - `PK11ThreadSafe`: only load PKCS#11 modules that are thread-safe, ie. that
+   support locking - either OS locking or NSS-provided locks. `boolean`,
+   defaults to `false`.
+ - `PKIXVerify`: Enable PKIX verify rather than the old cert library, to
+   verify certificates. `boolean`, defaults to `false`.
+ - `readOnly`: To open the databases in read-only mode, set this flag to
+   true. `boolean`, defaults to `false`.
+ - `removeSunProvider`: Remove the Sun crypto provider. `boolean`, defaults
+   to `false`.
+
+### `PasswordCallback` handlers
+
+In order to authenticate against a PKCS#11 token or to the internal
+certificate store, it is necessary to select a `PasswordCallback` handler.
+By default this is a console-based `PasswordCallback` handler. This prompts
+the user for the password via the Console. However, this is not appropriate
+in all scenarios.
+
+Developers are expected to extend and implement this as desired by their
+application.
+
+For example, to set a static `PasswordCallback` handler:
+
+```java
+CryptoManager cm = CryptoManager.getInstance();
+cm.setPasswordCallback(new Password("password".toCharArray()));
+```
+
+## Loading JSS via `java.security`
+
+You can directly add the `JSSProvider` by adding it to the `java.security`
+file:
+
+```properties
+security.provider.<n> = org.mozilla.jss.JSSProvider /path/to/jss.cfg
+```
+
+There are two ways use this `java.security` file: by directly installing
+it to the system or by using `-Djava.security.properties=/path/to/file`.
+Note that two equals signs may be used, in which case the system configuration
+is ignored and fully overridden by this file.
+
+`JSSProvider` behaves like the `SunPKCS11-NSS` provider, requiring a
+configuration file to initialize JSS (and the `CryptoManager` object).
+
+`jss.cfg` takes the same `InitializationValues` parameters, except in a
+properties file format.
+
+### JSS Config
+
+| property                           | `InitializationValues` member name  |
```

Comment:
Some properties are mapped to `CryptoManager` method, so "InitializationValues member name" is a bit inaccurate. How about "Mapped to"?